lib/SymbolGraphGen/Symbol.cpp

```diff
@@ -597,52 +634,70 @@ void Symbol::serialize(llvm::json::OStream &OS) const {
   });
 }
 
+swift::DeclName Symbol::getName(const Decl *D) const {
+  if (const auto *ED = dyn_cast<ExtensionDecl>(D)) {
+    return ED->getExtendedNominal()->getName();
+  } else {
+    return cast<ValueDecl>(D)->getName();
+  }
+}
+
 void
 Symbol::getPathComponents(SmallVectorImpl<PathComponent> &Components) const {
+  const ValueDecl *Decl = nullptr;
+  if (const auto *ED = dyn_cast<ExtensionDecl>(D)) {
+    Decl = ED->getExtendedNominal();
+  } else if (const auto *VD = dyn_cast<ValueDecl>(D)) {
+    Decl = VD;
+  }
+
   // Note: this is also used for sourcekit's cursor-info request, so can be
   // called on local symbols too. For such symbols, the path contains all parent
   // decl contexts that are currently representable in the symbol graph,
   // skipping over the rest (e.g. containing closures and accessors).
 
-  auto collectPathComponents = [&](const ValueDecl *Decl,
-                                   SmallVectorImpl<PathComponent> &DeclComponents) {
-    // Collect the spellings, kinds, and decls of the fully qualified identifier
-    // components.
-    while (Decl && !isa<ModuleDecl>(Decl)) {
-      SmallString<32> Scratch;
-      Decl->getName().getString(Scratch);
-      if (supportsKind(Decl->getKind()))
-        DeclComponents.push_back({Scratch, getKind(Decl).first, Decl});
-
-      // Find the next parent.
-      auto *DC = Decl->getDeclContext();
-      while (DC && DC->getContextKind() == DeclContextKind::AbstractClosureExpr)
-        DC = DC->getParent();
-      if (DC) {
-        if (const auto *Nominal = DC->getSelfNominalTypeDecl()) {
-          Decl = Nominal;
-        } else {
-          Decl = dyn_cast_or_null<ValueDecl>(DC->getAsDecl());
+  auto collectPathComponents =
```

Comment:
was this formatted with clang format? In general it is best to do formatting with git clang-format

Reply:
Sorry about that. I always formatted as suggested in the [guide](https://github.com/apple/swift/blob/main/docs/HowToGuides/FAQ.md#how-do-i-format-my-changes), so I'm not sure what happened there exactly. However, if I reformat the entire document now, it keeps the formatting as I committed it before. Thus, I guess git clang-format noticed the faulty indentation in one of the lines I edited and then expanded the re-format to the whole function-definition. Only changing the indentation of one line wouldn't make any sense either.

Reply:
It's completely fine, I forgot that it keeps a bit of context around the modifications.

Reply:
Resolved in 37f4c983c382991edd2c776e98250a9ce3a96d0e. I had to add a little more type assertions to make your code compile, though.